pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestWatcherReceivesNoopCheckpoints failed

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestWatcherReceivesNoopCheckpoints [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9162772?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9162772?buildTab=artifacts#/) on master @ [b89fa2cc4bc1fb9447ab1009b34b6f354f9618f0](https://github.com/cockroachdb/cockroach/commits/b89fa2cc4bc1fb9447ab1009b34b6f354f9618f0):


```
E230321 06:01:45.483610 3563854 jobs/adopt.go:225  [T1,n1] 1443  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:45.513442 3569529 jobs/adopt.go:261  [T1,n1] 1444  job 102: resuming execution
E230321 06:01:45.553120 3569529 jobs/adopt.go:225  [T1,n1] 1445  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:45.593657 3574522 jobs/adopt.go:261  [T1,n1] 1446  job 102: resuming execution
E230321 06:01:45.636754 3574522 jobs/adopt.go:225  [T1,n1] 1447  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:45.690172 3580391 jobs/adopt.go:261  [T1,n1] 1448  job 102: resuming execution
E230321 06:01:45.718916 3580391 jobs/adopt.go:225  [T1,n1] 1449  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:45.758229 3584758 jobs/adopt.go:261  [T1,n1] 1450  job 102: resuming execution
E230321 06:01:45.838108 3584758 jobs/adopt.go:225  [T1,n1] 1451  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:45.894760 3590274 jobs/adopt.go:261  [T1,n1] 1452  job 102: resuming execution
E230321 06:01:45.931624 3590274 jobs/adopt.go:225  [T1,n1] 1453  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:45.965331 3594968 jobs/adopt.go:261  [T1,n1] 1454  job 102: resuming execution
E230321 06:01:45.990390 3594968 jobs/adopt.go:225  [T1,n1] 1455  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:46.019786 3598906 jobs/adopt.go:261  [T1,n1] 1456  job 102: resuming execution
E230321 06:01:46.045357 3598906 jobs/adopt.go:225  [T1,n1] 1457  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:46.078729 3603730 jobs/adopt.go:261  [T1,n1] 1458  job 102: resuming execution
E230321 06:01:46.120591 3603730 jobs/adopt.go:225  [T1,n1] 1459  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:46.170908 3609978 jobs/adopt.go:261  [T1,n1] 1460  job 102: resuming execution
E230321 06:01:46.215022 3609978 jobs/adopt.go:225  [T1,n1] 1461  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:46.261110 3615655 jobs/adopt.go:261  [T1,n1] 1462  job 102: resuming execution
I230321 06:01:46.289962 3616574 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 1463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
E230321 06:01:46.306735 3615655 jobs/adopt.go:225  [T1,n1] 1464  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:46.334713 3620936 jobs/adopt.go:261  [T1,n1] 1465  job 102: resuming execution
E230321 06:01:46.362985 3620936 jobs/adopt.go:225  [T1,n1] 1466  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:46.394120 3625687 jobs/adopt.go:261  [T1,n1] 1467  job 102: resuming execution
E230321 06:01:46.426284 3625687 jobs/adopt.go:225  [T1,n1] 1468  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:46.461944 3630178 jobs/adopt.go:261  [T1,n1] 1469  job 102: resuming execution
E230321 06:01:46.502867 3630178 jobs/adopt.go:225  [T1,n1] 1470  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230321 06:01:46.541098 3635408 jobs/adopt.go:261  [T1,n1] 1471  job 102: resuming execution
    sqlwatcher_test.go:608: 
        	Error Trace:	pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test/pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/sqlwatcher_test.go:608
        	            				GOROOT/src/runtime/asm_amd64.s:1594
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestWatcherReceivesNoopCheckpoints
I230321 06:01:46.548934 3061498 testutils/testcluster/testcluster.go:144  [-] 1472  TestCluster quiescing nodes
I230321 06:01:46.550552 3084328 sql/stats/automatic_stats.go:507  [T1,n1] 1473  quiescing auto stats refresher
I230321 06:01:46.551057 3418647 jobs/registry.go:1494  [T1,n1] 1474  KEY VISUALIZER job 100: stepping through state succeeded
I230321 06:01:46.551146 3066200 server/start_listen.go:103  [T1,n1] 1475  server shutting down: instructing cmux to stop accepting
I230321 06:01:46.553276 3177803 rpc/context.go:2257  [T1,n1,rnode=1,raddr=127.0.0.1:36559,class=default,rpc] 1476  connection heartbeat loop ended with err: <nil>
W230321 06:01:46.550775 3080693 sql/sqlliveness/slinstance/slinstance.go:312  [n1] 1477  exiting heartbeat loop
W230321 06:01:46.560621 3080693 sql/sqlliveness/slinstance/slinstance.go:297  [n1] 1478  exiting heartbeat loop
E230321 06:01:46.560873 3080693 server/server_sql.go:493  [n1] 1479  failed to run update of instance with new session ID: node unavailable; try another peer
E230321 06:01:46.561982 3418647 jobs/adopt.go:472  [T1,n1] 1480  job 100: adoption completed with error job 100: could not mark as succeeded: job 100: select-job: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Table/15/1/100,/Table/15/1/101), [txn: 17df875c], [can-forward-ts] RPC error: node unavailable; try another peer
E230321 06:01:46.562100 3418647 jobs/registry.go:924  [T1,n1] 1481  error getting live session: node unavailable; try another peer
E230321 06:01:46.566385 3084431 jobs/registry.go:968  [T1,n1] 1482  error expiring job sessions: expire-sessions: failed to read query result: query execution canceled
E230321 06:01:46.566670 3084431 jobs/registry.go:976  [T1,n1] 1483  failed to serve pause and cancel requests: txn exec: context canceled
W230321 06:01:46.566796 3084431 jobs/registry.go:1052  [T1,n1] 1484  canceling all adopted jobs due to stopper quiescing
--- FAIL: TestWatcherReceivesNoopCheckpoints (6.57s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWatcherReceivesNoopCheckpoints.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25698

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestWatcherReceivesNoopCheckpoints [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9183235?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9183235?buildTab=artifacts#/) on master @ [78bb7696847e1d8abf5d6a199c6dd3414a568691](https://github.com/cockroachdb/cockroach/commits/78bb7696847e1d8abf5d6a199c6dd3414a568691):


```
I230322 06:17:26.907896 3588863 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 1162  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230322 06:17:26.984052 3093361 3@pebble/event.go:717  [n1,s1,pebble] 1163  [JOB 7] WAL created 000006
I230322 06:17:26.988110 3085677 1@server/server_sql.go:1668  [T1,n1] 1164  serving sql connections
I230322 06:17:26.989453 3085677 testutils/testcluster/testcluster.go:1348  [-] 1165  WaitForFullReplication
I230322 06:17:26.989565 3085677 testutils/testcluster/testcluster.go:1352  [-] 1166  WaitForFullReplication took: 352ns
I230322 06:17:27.020993 3598358 util/log/event_log.go:32  [T1,n1,client=127.0.0.1:41414,user=root] 1167 ={"Timestamp":1679465847006209505,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.rangefeed.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.rangefeed.enabled","Value":"true"}
I230322 06:17:27.043464 3598358 util/log/event_log.go:32  [T1,n1,client=127.0.0.1:41414,user=root] 1168 ={"Timestamp":1679465847025504392,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.closed_timestamp.target_duration\" = '100ms'","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.closed_timestamp.target_duration","Value":"00:00:00.1"}
I230322 06:17:27.072942 3094373 kv/kvserver/replica_rangefeed.go:703  [T1,n1,s1,r10/1:/Table/{6-7}] 1169  RangeFeed closed timestamp 1679465844.037201996,0 is behind by 3.035721539s
I230322 06:17:27.135766 3598358 5@util/log/event_log.go:32  [T1,n1,client=127.0.0.1:41414,user=root] 1170 ={"Timestamp":1679465847049452353,"EventType":"create_table","Statement":"CREATE TABLE defaultdb.public.t ()","Tag":"CREATE TABLE","User":"root","DescriptorID":104,"TableName":"defaultdb.public.t"}
I230322 06:17:27.210178 3616925 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:245  [-] 1171  established range feed over system.descriptors starting at time 1679465847.048748594,0
I230322 06:17:27.210345 3616925 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:304  [-] 1172  established range feed over system.zones starting at time 1679465847.048748594,0
I230322 06:17:27.215104 3616925 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:400  [-] 1173  established range feed over system.protected_ts_records starting at time 1679465847.048748594,0
I230322 06:17:27.229898 3617358 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 1174  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    sqlwatcher_test.go:608: 
        	Error Trace:	pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test/pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/sqlwatcher_test.go:608
        	            				GOROOT/src/runtime/asm_amd64.s:1594
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestWatcherReceivesNoopCheckpoints
I230322 06:17:27.799336 3085677 testutils/testcluster/testcluster.go:144  [-] 1175  TestCluster quiescing nodes
W230322 06:17:27.799706 3611728 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 1176  failed to gc transaction record: could not GC completed transaction anchored at /Table/3/1/104/2/1: node unavailable; try another peer
I230322 06:17:27.799979 3453599 jobs/registry.go:1494  [T1,n1] 1178  KEY VISUALIZER job 100: stepping through state succeeded
I230322 06:17:27.800376 3090942 server/start_listen.go:103  [T1,n1] 1179  server shutting down: instructing cmux to stop accepting
I230322 06:17:27.804312 3110969 sql/stats/automatic_stats.go:507  [T1,n1] 1180  quiescing auto stats refresher
I230322 06:17:27.805802 3190739 rpc/context.go:2245  [T1,n1,rnode=1,raddr=127.0.0.1:44013,class=default,rpc] 1181  connection heartbeat loop ended with err: <nil>
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177  job 102: running execution encountered retriable error: non-cancelable: context canceled
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +(1) attached stack trace
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  -- stack trace:
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  | 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1572
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  | 	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:467
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func2
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  | 	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:385
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  | runtime.goexit
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +  | 	GOROOT/src/runtime/asm_amd64.s:1594
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +Wraps: (2) non-cancelable
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +Wraps: (3) context canceled
I230322 06:17:27.799884 3562247 jobs/registry.go:1574  [T1,n1] 1177 +Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errors.errorString
E230322 06:17:27.806583 3562247 jobs/adopt.go:472  [T1,n1] 1182  job 102: adoption completed with error non-cancelable: context canceled
W230322 06:17:27.806855 3106797 sql/sqlliveness/slinstance/slinstance.go:312  [n1] 1183  exiting heartbeat loop
W230322 06:17:27.806919 3106797 sql/sqlliveness/slinstance/slinstance.go:297  [n1] 1184  exiting heartbeat loop
E230322 06:17:27.806966 3106797 server/server_sql.go:503  [n1] 1185  failed to run update of instance with new session ID: node unavailable; try another peer
E230322 06:17:27.809480 3111275 jobs/registry.go:976  [T1,n1] 1186  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: failed to read query result: query execution canceled
W230322 06:17:27.809604 3111275 jobs/registry.go:1052  [T1,n1] 1187  canceling all adopted jobs due to stopper quiescing
E230322 06:17:27.809765 3510324 jobs/registry.go:924  [T1,n1] 1188  error getting live session: node unavailable; try another peer
--- FAIL: TestWatcherReceivesNoopCheckpoints (6.29s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWatcherReceivesNoopCheckpoints.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/changefeedccl/changefeedbase/settings.go
+++ b/pkg/ccl/changefeedccl/changefeedbase/settings.go
@@ -224,7 +224,7 @@ var UseMuxRangeFeed = settings.RegisterBoolSetting(
 	settings.TenantWritable,
 	"changefeed.mux_rangefeed.enabled",
 	"if true, changefeed uses multiplexing rangefeed RPC",
-	util.ConstantWithMetamorphicTestBool("changefeed.mux_rangefeed.enabled", false),
+	false,
 )
 
 // EventConsumerWorkers specifies the maximum number of workers to use when

--- a/pkg/kv/kvclient/rangefeed/BUILD.bazel
+++ b/pkg/kv/kvclient/rangefeed/BUILD.bazel
@@ -22,7 +22,6 @@ go_library(
         "//pkg/roachpb",
         "//pkg/settings",
         "//pkg/settings/cluster",
-        "//pkg/util",
         "//pkg/util/admission/admissionpb",
         "//pkg/util/ctxgroup",
         "//pkg/util/hlc",

--- a/pkg/kv/kvclient/rangefeed/rangefeed.go
+++ b/pkg/kv/kvclient/rangefeed/rangefeed.go
@@ -24,7 +24,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
-	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -267,7 +266,9 @@ func (f *RangeFeed) Close() {
 // will be reset.
 const resetThreshold = 30 * time.Second
 
-var useMuxRangeFeed = util.ConstantWithMetamorphicTestBool("use-mux-rangefeed", false)
+// Disable mux rangefeed on 23.1 release.
+// TODO(yevgeniy): Re-enable with metamorphic constant.
+var useMuxRangeFeed = false
 
 // run will run the RangeFeed until the context is canceled or if the client
 // indicates that an initial scan error is non-recoverable.

